devlib/derived/__init__.py

```diff
@@ -12,6 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+from devlib.instrument import MeasurementType, MEASUREMENT_TYPES
+
+
```

Comment:
> The notion of an InstrumentChannel doesn't really make
sense in the context of DerivedMeasurments, which are not directly
measured on the target.

Hmm.. but `DerivedEnergyMeasurements` does derive per-channel metrics, so we technically we lose (although it's actually in the `DerivedMetric`'s `name`) a meaningful mapping between metric and channel there, right? I don't know if that actually matters at all...

Reply:
Yes, that's true, but in the general case, there won't necessarily be a `DerivedMetric` --> `InstrumentChannel` mapping, as a metric may be derived from multiple channels, or from auxiliary output. 

I have considered adding a `channels` attribute that would default to an empty list, but could be overwritten to identify the channels that were involved in deriving the metric. However, being unable to think of a good use for it, I decided to err on the side of simplicity.

I'm open to adding that, you think it would be useful.

Reply:
Yeah, I was thinking of an optional `channel` attribute like you suggested (and a list is better), but you're right it's probably unnecessary complexity unless we know a use case. Can always add it later if one arises.